config/webpack.config.common.ts

```diff
@@ -0,0 +1,52 @@
+import { Configuration } from "webpack";
+import { join, resolve } from "path";
+import HtmlWebpackPlugin from "html-webpack-plugin";
+
+export const config: Configuration = {
+  entry: resolve("src", "index.tsx"),
+  output: {
+    path: resolve("build"),
+    filename: "[name].[chunkhash].bundle.js",
+    clean: true,
+  },
+  resolve: {
+    extensions: [".tsx", ".ts", ".js"],
+  },
+  module: {
+    rules: [
+      {
+        test: /\.(js|jsx)$/,
+        exclude: /node_modules/,
+        use: ["babel-loader"],
+      },
+      {
+        test: /\.(ts|tsx)$/,
+        exclude: /node_modules/,
+        use: ["ts-loader"],
+      },
+      {
+        test: /\.(jpg|jpeg|png|gif|mp3|svg)$/,
+        use: ["file-loader"],
+      },
+    ],
+  },
+  plugins: [
+    new HtmlWebpackPlugin({
+      template: join("src", "index.html"),
+    }),
+  ],
+  optimization: {
```

Comment:
That would, probably, make more sense for production mode only.

Reply:
I don't agree here. Having a separate vendor chunk is useful in dev mode as well. It's better to keep the two modes as similar as possible to avoid prod bugs which not be reproduced on dev mode.
Also, it's a generally accepted practice to have a separate vendor chunk rather then build everything in one file